### ISSUE TITLE
--env argument is only accepted when running a lane

### ISSUE THREAD

**Issue by liamnichols** [fastlane/fastlane#16077]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
This isn't really a bug report, since I think that it's potentially implemented as intended however today i noticed that using `fastlane run my_action --env jenkins` results in the following error:

> invalid option: --env

We've just started using dotenv to organise all the environment variables along with overrides that are specific for different environments as described here: https://docs.fastlane.tools/best-practices/keys/#dotenv.

Usually this isn't a problem since we run a lane directly, but we have a custom action for various tasks and sometimes I need to test them locally via the `run` command, this is where I hit the roadblock.

As a temporary work around, I can just wrap the action in a lane however I want to know what the expected behavior is and potentially correct it if somebody agrees.

First time .env files are loaded is in **cli_tools_distributor.rb**, right at the very beginning

https://github.com/fastlane/fastlane/blob/099fd43ecc70f862e5346b61c80d89f8ec61038b/fastlane/lib/fastlane/cli_tools_distributor.rb#L62-L66

The reasoning in the comment makes sense, we'd like to hide non-essential output on the CI so I would like to define `FASTLANE_HIDE_CHANGELOG` in my `.env.jenkins` file, however because we pass `nil` into this method, only variables from `.env` and `.env.default` are loaded at this point.

Its then not until **lane_manager.rb** where we call `load_env_vars` again but this time, we include the value of `--env` passed in via the command line arguments:

https://github.com/fastlane/fastlane/blob/0c6a6d24098b9f6d7b16e4328a2048c6da4fa8df/fastlane/lib/fastlane/lane_manager.rb#L51

The above explains why I'm not able to pass `--env` when using `fastlane run [action]`, but my question is: is this expected?

Its not immediately obvious to me why we don't make `--env` as a `global_option` like `--verbose` is? 

<img width="1000" alt="Screenshot 2020-02-25 at 12 46 58" src="https://user-images.githubusercontent.com/482871/75248925-0c083c80-57cd-11ea-8863-375413e0942a.png">

https://github.com/fastlane/fastlane/blob/f32b007ff45e648b37b6c9c2037ac481f36b7780/fastlane/lib/fastlane/commands_generator.rb#L102

Sorry for the strange format of this ticket... its more of a question than a bug report... if one of the contributors can help me out here, I'm happy to work on changing the behaviour but I'm guessing that there is more to consider

Thanks fro all the hard work 🙏

**Comment by janpio** [comment]
> Its not immediately obvious to me why we don't make `--env` as a `global_option` like `--verbose` is?

I agree, might have been an oversight or no one asked for it ever.

Unless someone comes up with a good reason why this is a bad idea, we could very well introduce this.

**Comment by liamnichols** [comment]
Thanks for the confirmation @janpio... I've got some spare time today so might dig in a little deeper and see how hard it would be to do this while we wait to see if anybody has any other objections

**Comment by liamnichols** [comment]
Pushed a draft to #16096... Needs a bit of input before it's ready though, it may well be the wrong approach still

**Comment by liamnichols** [comment]
Does this need the **status: has-pr** label to keep @fastlane-bot happy @janpio?

**Comment by janpio** [comment]
Not sure if that helps tbh - but replying does ;)
